```bash
$ alohomora
Traceback (most recent call last):
  File "/Users/skemper/virtualenvs//alohomora-dev3/bin/alohomora", line 11, in <module>
    load_entry_point('alohomora', 'console_scripts', 'alohomora')()
  File "/Users/skemper/virtualenvs/alohomora-dev3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 490, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/Users/skemper/virtualenvs/alohomora-dev3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2853, in load_entry_point
    return ep.load()
  File "/Users/skemper/virtualenvs/alohomora-dev3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2444, in load
    return self.resolve()
  File "/Users/skemper/virtualenvs/alohomora-dev3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2450, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/Users/skemper/Code/alohomora/alohomora/main.py", line 37, in <module>
    import alohomora.req
  File "/Users/skemper/Code/alohomora/alohomora/req.py", line 68, in <module>
    if not get_u2f_devices():
  File "/Users/skemper/Code/alohomora/alohomora/req.py", line 59, in get_u2f_devices
    devices = u2f.list_devices()
NameError: name 'u2f' is not defined
```

This happens when installing *just* `alohomora`, and then running it.  I believe the problem is with req.py, line 68.  That conditional should look like:

```python
if U2F_SUPPORT and not get_u2f_devices():
```

Alternately we could move the import below the definition of `get_u2f_devices()` and then do the test within the import statement.  Thoughts?

--- a/README.md
+++ b/README.md
@@ -14,6 +14,26 @@ but the choice is yours.
 
     pip install alohomora
 
+### Optional U2F support installation
+
+Alohomora has optional U2F support, which can be installed alongside alohomora using pip:
+
+    pip install alohomora[u2f]
+
+Please note that this requires a few OS level packages to be installed.
+For Centos 7, it requires the following:
+
+    yum groupinstall 'Development Tools'
+    yum install python[2,3]-devel
+    yum install libusbx-devel
+    yum install systemd-devel
+
+For Debian based systems, you'll need the following:
+
+    apt install build-essential
+    apt install python[2,3]-dev
+    apt install libusb-1.0-0-dev
+    apt install libudev-dev
 
 ## Basic Configuration
 

--- a/alohomora/req.py
+++ b/alohomora/req.py
@@ -38,12 +38,13 @@ from bs4 import BeautifulSoup
 
 import alohomora
 
-U2F_SUPPORT = True
+U2F_SUPPORT = False
 try:
     from u2flib_host import u2f, exc
     from u2flib_host.constants import APDU_USE_NOT_SATISFIED, APDU_WRONG_DATA
+    U2F_SUPPORT = True
 except ImportError:
-    U2F_SUPPORT = False
+    pass
 
 try:
     input = raw_input #pylint: disable=redefined-builtin,invalid-name
@@ -65,8 +66,12 @@ def get_u2f_devices():
     return devices
 
 
-if not get_u2f_devices():
-    U2F_SUPPORT = False
+if U2F_SUPPORT:
+    try:
+        get_u2f_devices()
+        U2F_SUPPORT = True
+    except: # pylint: disable=bare-except
+        U2F_SUPPORT = False
 
 
 class DuoDevice(object):

--- a/setup.py
+++ b/setup.py
@@ -34,6 +34,8 @@ setup(
         "boto3>=1.3.1",
         "beautifulsoup4>=4.5.1",
         "requests>=2.11.1",
-        "python-u2flib-host>=3.0.3",
     ],
+    extras_require={
+        "u2f": ["python-u2flib-host>=3.0.3"]
+    }
 )

